src/expr/src/expr/mod.rs

```diff
@@ -128,7 +136,7 @@ impl RowExpression {
 
     pub fn eval(&mut self, row: &Row, data_types: &[DataType]) -> Result<ArrayRef> {
         let input = DataChunk::from_rows(slice::from_ref(row), data_types)?;
-        self.expr.eval(&input)
+        self.expr.wrapping_eval(&input)
```

Comment:
Is `eval()` useless now?

Reply:
eval is the implementation, and wrapping_eval add some common checks.

Reply:
`wrapping` sounds like a numeric overflow wrapping to me. 😢 Name it `checked_eval` or something?

Reply:
wrapping is just eval with a wrapper :)

Currently we do check here, but I'm thinking about something else, such as tracing span, and the check may also be updated to debug_assert! if we have some benchmarks.